### PR TITLE
Run CI workflow on macOS, Windows, and Linux

### PR DESCRIPTION
**Pull request by externl**
We don't run template testing on Windows. See #3976

### CODE REVIEW

**Review by bernardnormier — APPROVED** [review]

File: .github/actions/test/action.yaml [hunk 1]
```diff
@@ -7,5 +7,5 @@ runs:
     run: cargo test --manifest-path tools/slicec-cs/Cargo.toml
     shell: bash
   - name: 🧪 Test
-    run: dotnet test --no-build --verbosity normal --blame-hang-timeout 8m
+    run:  dotnet test --no-build --verbosity normal --blame-hang-timeout 10s tests/IceRpc.Tests/IceRpc.Tests.csproj
```
Comment (InsertCreativityHere) [inline .github/actions/test/action.yaml:10]: Is this extra space intentional? It's fine if you want it for some reason. Just double-checking.
```suggestion
    run: dotnet test --no-build --verbosity normal --blame-hang-timeout 10s tests/IceRpc.Tests/IceRpc.Tests.csproj
```

Comment (externl) [reply .github/actions/test/action.yaml]: Not intentional, just a bad copy and paste.

File: .github/actions/test/action.yaml [hunk 2]
```diff
@@ -7,5 +7,13 @@ runs:
     run: cargo test --manifest-path tools/slicec-cs/Cargo.toml
     shell: bash
   - name: 🧪 Test
-    run: dotnet test --no-build --verbosity normal --blame-hang-timeout 8m
+    # See https://github.com/microsoft/vstest/issues/2080#issuecomment-539879345
+    run: dotnet test --no-build --verbosity normal --blame-hang-timeout 10s --logger:"console;noprogress=true"
```
Comment (InsertCreativityHere) [inline .github/actions/test/action.yaml:11]: How long does a single test usually take? 10s feels on the short side to me, but to be fair, I don't really have a scale for how quickly we expect these tests to complete...

**Review by InsertCreativityHere — APPROVED** [review]
Looks good to me!

File: .github/actions/test/action.yaml (hunk 2 above)
Comment (externl) [reply .github/actions/test/action.yaml:11]: It's for a single unit test. So it should be much faster than 10s

File: .github/workflows/ci.yml [hunk 3]
```diff
@@ -6,10 +6,19 @@ on:
...
 jobs:
   build_and_test:
-    runs-on: ubuntu-latest
-    timeout-minutes: 10
+    strategy:
+      fail-fast: false
+      matrix:
+        os: [ubuntu-latest, macos-latest, windows-latest]
+    runs-on: ${{ matrix.os }}
+    timeout-minutes: 20
```
Comment (pepone) [inline .github/workflows/ci.yml:21]: Should we revert this to 10 minutes?

**Comment by externl** [comment]
In the interest of getting this PR merged I've removed macOS from the matrix.

**Review by pepone — APPROVED** [review]

**Pull request merged**